### PR TITLE
fix: fix VS CMake generator finder 

### PR DESCRIPTION
**Pull request by diefbell-grabcad**
If a newer version of Visual Studio build tools (>= 2019) is installed then the automatic build tools finder fails because `cmake -G` outputs:
```
Generators
* Visual Studio 17 2022        = Generates Visual Studio 2022 project files.
                                 Use -A option to specify architecture.
  Visual Studio 16 2019        = Generates Visual Studio 2019 project files.
                                 Use -A option to specify architecture.
  Visual Studio 15 2017 [arch] = Generates Visual Studio 2017 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 14 2015 [arch] = Generates Visual Studio 2015 project files.
```

and the function `GET_CMAKE_VS_GENERATOR` doesn't recognise them because it expects the ` [arch]` suffix. 

This fixes:
    - Above issue;
    - Issue where one of the build tools is prefixed by `* ` (I think that means it's the system default or something? Idk;)
    - For the build tools that don't end with ` [arch]`, it appends `-A arch` to the generator string in the argumentBuilder (kinda janky).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cmake-ts",
-  "version": "0.3.0",
+  "version": "0.4.0",
```
Comment (aminya) [inline package.json:3]: Please keep the version as is. We can publish it as a patch

**Review by aminya — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (diefbell-grabcad) [reply package.json:3]: So bump patch version? Or just leave as is?

Comment (aminya) [reply package.json:3]: Leave as is. I'll handle the version bunp

Comment (diefbell-grabcad) [reply package.json]: Done

**Comment by diefbell-grabcad** [comment]
> I had rebased this into a single commit. Any reason to undo that? Could you rebase and squash the changes?

I have no idea how that happened. I can try just creating a new branch. I think because my branch was off your fork originally.

**Comment by aminya** [comment]
Sounds good. Here's the branch I rebased:
https://github.com/EmbeddedEnterprises/cmake-ts/tree/vs-cmake-generator

**Comment by diefbell-grabcad** [comment]
Created https://github.com/EmbeddedEnterprises/cmake-ts/pull/32 without mess un-squash

**Pull request closed without merging**